### PR TITLE
restore old `ExpandableBlock` api

### PR DESCRIPTION
**Pull request by mayank99**
## Changes

undid the breaking change from #1354 and brought back `title`/`caption`/`endIcon` props while also supporting the new subcomponents added in that PR.

renamed `ExpandableBlock.Header` to `ExpandableBlock.Trigger` as it makes more sense.

these two snippets are now equivalent:
```jsx
<ExpandableBlock title='click me' caption='or not' endIcon={<svg>...</svg>}>
  content in block
</ExpandableBlock>
```
```jsx
<ExpandableBlock.Wrapper>
  <ExpandableBlock.Trigger>
    <ExpandableBlock.ExpandIcon />
    <ExpandableBlock.LabelArea>
      <ExpandableBlock.Title>click me</ExpandableBlock.Title>
      <ExpandableBlock.Caption>or not</ExpandableBlock.Caption>
    </ExpandableBlock.LabelArea>
    <ExpandableBlock.EndIcon><svg>...</svg></ExpandableBlock.EndIcon>
  </ExpandableBlock.Trigger>
  <ExpandableBlock.Content>content in block</ExpandableBlock.Content>
</ExpandableBlock.Wrapper>
```

## Testing

updated one story and added one unit test.

## Docs

updated old changeset. updated documentation page.

will update migration guide too.

### CODE REVIEW

**Review by gretanausedaite — APPROVED** [review]

**Review by rohan-kadkol — COMMENTED** [review]
Having some website examples that still use the old API would be good.

**Comment by mayank99** [comment]
> Having some website examples that still use the old API would be good.

Updated. [`bf00897`](https://github.com/iTwin/iTwinUI/pull/1502/commits/bf00897be146765d9a3a5afc7c46f69eb6016f7d)

File: apps/website/src/pages/docs/expandableblock.mdx [hunk 1]
```diff
@@ -14,17 +14,22 @@ import * as AllExamples from 'examples';
...
 
 The expandable block has a label reflecting its content. On click, it expands to reveal more content.
 
-## Variants
+## Usage
+
+The expandable block can be used in two ways:
+
+- Directly using `ExpandableBlock` along with `title`, `caption`, `endIcon` and children. This pattern can suffice for simple cases.
+- Using with subcomponents (`ExpandableBlock.Wrapper`, `ExpandableBlock.Trigger`, `ExpandableBlock.Content`, etc). This pattern is more verbose but allows full customization over all the parts.
```
Comment (rohan-kadkol) [inline apps/website/src/pages/docs/expandableblock.mdx:28]:
```suggestion
- Using subcomponents (`ExpandableBlock.Wrapper`, `ExpandableBlock.Trigger`, `ExpandableBlock.Content`, etc). This pattern is more verbose but allows full customization over all the parts.
```

File: apps/website/src/pages/docs/expandableblock.mdx [hunk 2]
```diff
@@ -70,9 +75,9 @@ An expandable block can be disabled by using the `disabled` prop.
   <AllExamples.ExpandableBlockDisabledExample client:load />
 </LiveExample>
 
-## Usage
+## Guidelines
```
Comment (rohan-kadkol) [inline apps/website/src/pages/docs/expandableblock.mdx:78]: We use *"Usage"* in other places on the website. So should we do the same here? Why did you change to *"Guidelines"*?

Comment (mayank99) [reply apps/website/src/pages/docs/expandableblock.mdx:78]: Because "Usage" doesn't make sense here. Usage is more about *how* to use it (like the section near the top of this file). I'm quite unhappy with how most pages on the website are structured.

**Review by rohan-kadkol — APPROVED** [review]
